docker/docker_tools.py

```diff
@@ -248,12 +276,23 @@ def build(
             print(f"Unable to run git: {e} Installing...")
             xqt("sudo apt-get install -y --no-install-recommends git")
 
+        # Check if Clone-All Flag is set
+        if clone_all != "RunestoneInteractive":
+            clone_bs = clone_all
+            clone_rc = clone_all
+            clone_rs = clone_all
+
         # Are we inside the Runestone repo?
         if not (wd / "nginx").is_dir():
             change_dir = True
             # No, we must be running from a downloaded script. Clone the runestone repo.
             print("Didn't find the runestone repo. Cloning...")
-            xqt("git clone https://github.com/RunestoneInteractive/RunestoneServer.git")
+            try:
+                # Check if possible to clone RunestoneServer with Custom Repo
+                xqt("export GIT_TERMINAL_PROMPT=0 && git clone https://github.com/%s/RunestoneServer.git"%clone_rs)
+            except subprocess.CalledProcessError as e:
```

Comment:
I suggest omitting this -- if the repo clone fails, exit with an error. Otherwise, the user won't see the difference in all the output produced by a build.

Reply:
I decided to change the except to perform a "sys.exit()" with a message stating there was an error cloning the system a remote repository with specific username.

Reply:
Good. I think that `sys.exit()` requires an int, though? You can use [click.echo](https://click.palletsprojects.com/en/8.0.x/quickstart/#echoing) to display a message.

Reply:
```sys.exit()``` has an optional argument. If nothing is specified within the function or ```None``` value is passed then the default exit code is 0. I know you can pass a int between 0-127 to refer to different error types. I passed a string object which prints out to STDERR (Terminal) and returns an exit code 1. 

It worked during testing by providing it non-existing usernames/repositories to see if ERROR reported out correctly.

Reference:
[Python sys module Documentation](https://docs.python.org/3/library/sys.html)

Reply:
Thanks, I learned something (sys.exit(string))!